src/ucp/wireup/select.c

```diff
@@ -1120,8 +1165,38 @@ ucp_wireup_add_rma_bw_lanes(ucp_wireup_select_ctx_t *select_ctx)
             continue;
         }
 
-        ucp_wireup_add_bw_lanes(select_ctx, &bw_info,
-                                context->mem_type_access_tls[mem_type]);
+        ucp_wireup_bw_info_set_iface_flags(context, &bw_info, 0);
+        status = ucp_wireup_add_bw_lanes(select_ctx, &bw_info,
+                                         context->mem_type_access_tls[mem_type]);
+        ucp_wireup_bw_info_unset_iface_flags(context, &bw_info, 0);
+        if ((status == UCS_OK) ||
+            /* If an exact rndv mode requested, we don't have
+             * a possibility to use another flag */
+            (context->config.ext.rndv_mode != UCP_RNDV_MODE_AUTO)) {
+            continue;
+        }
+
+        /* Try to add RMA BW lane with GET ZCOPY iface flag only */
+        ucp_wireup_bw_info_set_iface_flags(context, &bw_info,
```

Comment:
so we can add some lanes which support only get and others which support only put?
Will it work with rndv multirail?

Reply:
> Will it work with rndv multirail?

you're right - it doesn't work
how should we behave in this case? should we select lanes with the same iface_flag (i.e. either get-put, or get, or put)?

Reply:
in case of auto, I'd try to find get supporting tls, and fallback to put if not found.

Reply:
> in case of auto, I'd try to find get supporting tls, and fallback to put if not found.

that's what we do here right now, but it may happen that we can take 2 different lanes (GET only and PUT only)